### PR TITLE
Add the first test script

### PR DESCRIPTION
**Pull request by mipo1357**
Added tests directory in examples, added simple tests for metadata. Vitest is used for testing framework.
You can run it with `npm run test`.

### CODE REVIEW

File: examples/tests/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "tests",
+  "private": true,
+  "version": "0.0.0",
+  "type": "module",
+  "scripts": {
+    "dev": "vite",
```
Comment (jamie-lemon) [inline examples/tests/package.json:7]: not sure why we need vite here?

File: examples/tests/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    "preview": "vite preview",
+    "test": "vitest --run"
+  },
+  "devDependencies": {
+    "mupdf": "^0.2.1",
+    "typescript": "^5.2.2",
+    "vite": "^5.2.0",
```
Comment (jamie-lemon) [inline examples/tests/package.json:14]: not sure why we need vite here?

**Review by jamie-lemon — CHANGES_REQUESTED** [review]
Does vitest depend on vite?

When I looked vitest looked like a nice front-end package, but it seems you can run vitest without it I think.

**Comment by mipo1357** [comment]
The vitest example I was referring to was creating an environment from vite, so the code was still there. I deleted it.
I have confirmed that it does indeed work with vitest only.

**Pull request merged**